### PR TITLE
Expose status in get carrier servers and get carriers

### PR DESCRIPTION
**Pull request by ironBatista**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

#### Description
Added status Icon attributes to Carrier and CarrierServer entities.

### CODE REVIEW

**Comment by ironArt3mis** [comment]
Rebase completed.

File: library/Ivoz/Kam/Infrastructure/Kamailio/TrunksClient.php [hunk 1]
```diff
@@ -138,6 +138,10 @@ public function getLcrGatewayInfo($gw_id): array
          * }
          */
 
+        if (is_null($result->gw[0])) {
+            return [];
+        }
+
         return $result->gw[0];
```
Comment (mmadariaga) [inline library/Ivoz/Kam/Infrastructure/Kamailio/TrunksClient.php:145]: `return $result->gw[0] ?? [];`

File: library/Ivoz/Provider/Application/Service/Carrier/CarrierDtoAssembler.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+        $carrierServers = $this
+            ->carrierServerRepository
+            ->findByCarrierId(
+                (int) $entity->getId()
+            );
+
+        $carrierDto->addStatus(
+            new CarrierStatus(
+                true
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/Carrier/CarrierDtoAssembler.php:44]: Is this suppoused to be true when no carrier servers are found?

File: library/Ivoz/Provider/Application/Service/CarrierServer/StatusFromLcrGatewayInfo.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Ivoz\Provider\Application\Service\CarrierServer;
+
+use Ivoz\Kam\Domain\Model\TrunksLcrGateway\TrunksLcrGatewayInterface;
+use Ivoz\Kam\Domain\Service\TrunksClientInterface;
+use Ivoz\Kam\Infrastructure\Persistence\Doctrine\TrunksLcrGatewayDoctrineRepository;
+use Ivoz\Provider\Domain\Model\CarrierServer\CarrierServerDto;
+
+class StatusFromLcrGatewayInfo
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/CarrierServer/StatusFromLcrGatewayInfo.php:10]: I think that `GetStatusByCarrierServerDto` would be a better name

File: library/Ivoz/Provider/Application/Service/CarrierServer/StatusFromLcrGatewayInfo.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            );
+        } catch (\Exception $e) {
+            return null;
+        }
+
+        if (sizeof($lcrGatewayInfo) < 1) {
+            return null;
+        }
+
+        return $lcrGatewayInfo['state'];
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/CarrierServer/StatusFromLcrGatewayInfo.php:50]: ` return $lcrGatewayInfo['state'] ?? [];`

File: library/Ivoz/Provider/Domain/Model/Carrier/CarrierDto.php [hunk 5]
```diff
@@ -2,8 +2,13 @@
 
 namespace Ivoz\Provider\Domain\Model\Carrier;
 
+use Ivoz\Provider\Domain\Model\Carrier\CarrierStatus;
+
 class CarrierDto extends CarrierDtoAbstract
 {
+    /** @var CarrierStatus[]  */
+    private array $statusIcon = [];
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Carrier/CarrierDto.php:10]: This should not be an array, just CarrierStatus

Comment (mmadariaga) [reply library/Ivoz/Provider/Domain/Model/Carrier/CarrierDto.php:10]: Name it status, not statusIcon (Icon is a frontend concept)

File: library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerDto.php [hunk 6]
```diff
@@ -7,7 +7,7 @@ class CarrierServerDto extends CarrierServerDtoAbstract
     public const CONTEXT_STATUS = 'status';
 
     /** @var CarrierServerStatus[]  */
-    private array $status = [];
+    private array $statusIcon = [];
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerDto.php:10]: $status

File: library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerDto.php [hunk 7]
```diff
@@ -76,19 +79,19 @@ public function denormalize(array $data, string $context, string $role = ''): vo
 
     public function addStatus(CarrierServerStatus $status): static
     {
-        $this->status[] = $status;
+        $this->statusIcon[0] = $status;
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerDto.php:82]: why `[0]`?

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: library/Ivoz/Provider/Application/Service/Carrier/CarrierDtoAssembler.php (hunk 2 above)
Comment (ironBatista) [reply library/Ivoz/Provider/Application/Service/Carrier/CarrierDtoAssembler.php:44]: No, sorry, this is a logic error.

File: library/Ivoz/Provider/Application/Service/Carrier/CarrierDtoAssembler.php [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            )
+        );
+
+        if (sizeof($carrierServers) > 0) {
+            $carrierDto->addStatus(
+                new CarrierStatus(
+                    false
+                )
+            );
+        }
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/Carrier/CarrierDtoAssembler.php:55]:
```
$carrierDto->addStatus(
            new CarrierStatus(
                count($carrierServers) > 0
            )
        );
```

File: library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerDto.php [hunk 9]
```diff
@@ -6,8 +6,7 @@ class CarrierServerDto extends CarrierServerDtoAbstract
 {
     public const CONTEXT_STATUS = 'status';
 
-    /** @var CarrierServerStatus[]  */
-    private array $status = [];
+    private ?CarrierServerStatus $status = null;
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerDto.php:9]: $status is `CarrierServerStatus[] in CarrierServer`, and `CarrierServerStatus in Carrier`

File: library/Ivoz/Provider/Application/Service/CarrierServer/CarrierServerDtoAssembler.php [hunk 10]
```diff
@@ -33,40 +31,29 @@ public function toDto(
 
         $carrierServerDto = $entity->toDto($depth);
 
-        if (is_null($carrierServerDto->getId())) {
-            return $carrierServerDto;
-        }
+        $status = $this
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/CarrierServer/CarrierServerDtoAssembler.php:34]: This should go below `if (!$showStatus) ....`

File: library/Ivoz/Provider/Application/Service/CarrierServer/CarrierServerDtoAssembler.php [hunk 11]
```diff
@@ -33,40 +29,29 @@ public function toDto(
...
+                )
+            );
         }
 
          $carrierServerDto->addStatus(
              new CarrierServerStatus(
-                 (int) $status
+                 $status === 0
              )
          );
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/CarrierServer/CarrierServerDtoAssembler.php:56]: Missing early return in line 49 or 
} else { .... } return $carrierServerDto;

**Review by mmadariaga — CHANGES_REQUESTED** [review]

**Review by mmadariaga — APPROVED** [review]

**Comment by ironArt3mis** [comment]
Rebase completed.

**Pull request merged**